src/main/java/com/cloudbees/jenkins/support/configfiles/SecretHandler.java

```diff
@@ -77,13 +77,13 @@ public void characters(char[] ch, int start, int length) throws SAXException {
             }
         };
         Source src = new SAXSource(xr, new InputSource(new StringReader(FileUtils.readFileToString(xmlFile))));
-        File patchedFile = File.createTempFile("patched", ".xml");
-        Result res = new StreamResult(new FileOutputStream(patchedFile));
+        final ByteArrayOutputStream result = new ByteArrayOutputStream();
+        Result res = new StreamResult(result);
         Transformer transformer = TransformerFactory.newInstance().newTransformer();
         //omit xml declaration because of https://bugs.openjdk.java.net/browse/JDK-8035437
         transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
         transformer.transform(src, res);
 
-        return patchedFile;
+        return result.toString("UTF-8");
```

Comment:
is the factory guaranteed to write in `UTF-8`?
I think you need to be explicit by setting [encoding](https://docs.oracle.com/javase/7/docs/api/javax/xml/transform/OutputKeys.html#ENCODING)